internal/cmd/liquid.go

```diff
@@ -0,0 +1,416 @@
+	// I.e. Api currently only works when receiving service type cinder and not with liquid-cinder
+	reqServiceType := strings.TrimPrefix(serviceType, "liquid-")
+	var r gophercloud.Result
+	resp, err := limesAdminClient.Get(cmd.Context(), limesAdminClient.ServiceURL(fmt.Sprintf("liquid/service-usage-request?service_type=%s&project_id=%s", reqServiceType, projectID)), &r.Body, nil) //nolint:bodyclose
+	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
+	if err != nil {
+		return util.WrapError(err, "could not fetch service usage request body from limes")
+	}
+
+	var serviceUsageRequest *liquid.ServiceUsageRequest
+	err = r.ExtractInto(&serviceUsageRequest)
+	if err != nil {
+		return util.WrapError(err, "could not parse service usage request body response from limes")
+	}
+
+	provider, err := authenticate(cmd.Context())
+	if err != nil {
+		return util.WrapError(err, "could not authenticate with openstack")
+	}
+
+	var serviceUsageReport liquid.ServiceUsageReport
+	if compare || endpoint == "" {
+		serviceUsageReport, err = GetLiquidUsageReport(provider, liquidapi.ClientOpts{ServiceType: serviceType}, cmd.Context(), projectID, serviceUsageRequest, compare)
+		if err != nil {
+			return err
+		}
+	}
+
+	var localServiceusageReport liquid.ServiceUsageReport
+	if endpoint != "" {
+		localServiceusageReport, err = GetLiquidUsageReport(provider, liquidapi.ClientOpts{EndpointOverride: endpoint}, cmd.Context(), projectID, serviceUsageRequest, compare)
+		if err != nil {
+			return err
+		}
+	}
+
+	if compare {
+		diff := cmp.Diff(serviceUsageReport, localServiceusageReport)
+		if diff == "" {
+			fmt.Println("ServiceUsageReports are identical")
+		} else {
+			fmt.Println(diff)
+		}
+	}
+
+	return nil
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// LIQUID Set Quota
+
+type liquidSetQuotaCmd struct {
+	*cobra.Command
+
+	liquidQuotaOperationFlags
+}
+
+func newLiquidSetQuotaCmd() *liquidSetQuotaCmd {
+	liquidSetQuota := &liquidSetQuotaCmd{}
+	cmd := &cobra.Command{
+		Use:   "set-quota $SERVICE_TYPE $PROJECT_UUID",
+		Short: "Updates quota within a project if a liquid service",
+		Long:  "Updates quota within a project across specified resources of a liquid. This command requires a cloud-admin token.",
+		Args:  cobra.ExactArgs(2),
+		RunE:  liquidSetQuota.Run,
+	}
+
+	// Flags
+	doNotSortFlags(cmd)
+	liquidSetQuota.liquidQuotaOperationFlags.AddToCmd(cmd)
+
+	liquidSetQuota.Command = cmd
+	return liquidSetQuota
+}
+
+func (c *liquidSetQuotaCmd) Run(cmd *cobra.Command, args []string) error {
+	serviceType := args[0]
+	projectID := args[1]
+
+	endpoint := c.liquidQuotaOperationFlags.endpoint
+
+	if len(c.liquidQuotaOperationFlags.quotaValues) == 0 {
+		return errors.New("flag --quota_values is required")
+	}
+	var serviceQuotaRequest liquid.ServiceQuotaRequest
+	serviceQuotaRequest.Resources = make(map[liquid.ResourceName]liquid.ResourceQuotaRequest)
```

Comment:
I would like to see a validation here that quotas were provided for all resources, i.e. `serviceQuotaRequest.Resources` should contain entries for all resources in the ServiceInfo reported by the liquid.